mbedtls-sys/build/bindgen.rs

```diff
@@ -166,5 +185,11 @@ impl super::BuildConfig {
 
         let mod_bindings = self.out_dir.join("mod-bindings.rs");
         fs::write(mod_bindings, b"mod bindings;\n").expect("mod-bindings.rs I/O error");
+        
+        // compile static function wrappers with headers
+        fs::write(&static_wrappers_c_path, &headers).expect("write static_wrappers.c I/O error");
+        cc.file(&static_wrappers_c_path);
+        cc.compile("libstatic-wrappers.a");
```

Comment:
This shouldn't be done as part of the `bindgen` step. I think you're also missing the CFLAGS we're supposed to set when using `freestanding`.

Reply:
Moved these to `build.rs`